service/resources/sftp.py

```diff
@@ -0,0 +1,140 @@
+            return self.resp_error("Invalid Filename", resp)
+
+        data = req.bounded_stream.read()
+        file = open(local_file_path, "wb+")
+        file_written = file.write(data)
+        file.close()
+
+        if not isinstance(file_written, int):
+            return self.resp_error("Cannot write file", resp)
+
+        local_file_host_path = os.path.join(local_dir, 'host_'+file_name)
+        file_host = open(local_file_host_path, "w+")
+        file_host_written = file_host.write(sftp_info['HOST-KEY'])
+        file_host.close()
+
+        if not file_host_written:
+            if file_written:
+                try:
+                    os.remove(local_file_path)
+                #pylint: disable=broad-except
+                except Exception as exception:
+                    with sentry_sdk.configure_scope() as scope:
+                        scope.set_extra('remote_local_exception', str(exception))
+            return self.resp_error("Cannot write host file", resp)
+
+        sftp_put = None
+
+        if file_written:
+            try:
+                sftp_put = self.transfer_file(sftp_info, local_file_path,
+                                              remote_path, local_file_host_path)
+            #pylint: disable=broad-except
+            except Exception as exception:
+                msg_error = "Cannot transfer file: "+str(exception)
+                return self.resp_error(msg_error, resp)
+            finally:
+                os.remove(local_file_path)
+                os.remove(local_file_host_path)
+
+        if sftp_put:
+            msg = {'message': "File transferred successfully"}
+            resp.body = json.dumps(jsend.success(msg))
+            resp.status = falcon.HTTP_200
+            return True
+
+        return self.resp_error(msg_error, resp)
+
+    @staticmethod
+    def resp_error(msg, resp):
+        """ Sets response error """
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_extra('message', msg)
+            sentry_sdk.capture_message('sftp.resp_error', 'error')
+        resp.body = json.dumps(jsend.error(msg))
+        resp.status = falcon.HTTP_400
+
+    @staticmethod
+    def transfer_file(sftp_info, localpath,
+                      remotepath, file_host_path=None, cnopts=None):
+        """ Transfer file method """
+        sftp_host = sftp_info['HOST']
+        sftp_user = sftp_info['USER']
+        sftp_password = sftp_info['PASSWORD']
+
+        sftp_put = None
+
+        if not cnopts:
+            cnopts = pysftp.CnOpts()
+        if file_host_path:
+            cnopts.hostkeys.load(file_host_path)
+
+        sftp = pysftp.Connection(sftp_host, cnopts=cnopts,
+                                 username=sftp_user, password=sftp_password)
+
+        if sftp:
+            sftp.cd()
+            sftp.chdir(remotepath)
+            sftp_put = sftp.put(localpath, preserve_mtime=True)
+
+        return sftp_put
```

Comment:
Need to close the connection?

Reply:
updated in https://github.com/SFDigitalServices/sftp-microservice-py/pull/1/commits/80fa9b3adac54324235c8d773b257607f277aa55